repositoryref.go

```diff
@@ -0,0 +1,258 @@
+/*
+Copyright 2020 The Flux CD contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gitprovider
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+// TODO: Add equality methods for OrganizationRef and RepositoryRefs
+
+// OrganizationRef references an organization in a Git provider
+type OrganizationRef interface {
+	// String returns the HTTPS URL
+	fmt.Stringer
+
+	// GetDomain returns the URL-domain for the Git provider backend, e.g. gitlab.com or version.aalto.fi
```

Comment:
Should we replace `version.aalto.fi` with some imaginary/common alternative GitLab provider?